python/sglang/srt/models/deepseek_v2.py

```diff
@@ -708,7 +731,18 @@ def forward_absorb(
                 torch.bfloat16,
             )
         else:
-            attn_bmm_output = torch.bmm(attn_output.transpose(0, 1), self.w_vc)
+            if self.use_intel_amx_backend:
+                # See [Note] Align shapes of bmm inputs.
+                B = self.w_vc.size(0)
+                N = self.w_vc.size(1)
+                M = attn_output.size(0)
+                attn_bmm_output = torch.empty([B, M, N], dtype=attn_output.dtype)
```

Comment:
for this one you can also pass `attn_bmm_output` in non-contiguous form:

```
attn_bmm_output = torch.empty([M, B, N], dtype=attn_output.dtype).transpose_(0, 1)
```
and then you can skip attn_output = attn_bmm_output.transpose(0, 1).flatten(1, 2).

and i suppose when M != 1, the current code may trigger runtime error in self.o_proj as attn_output is not contiguous

but if you allocat attn_output as [M, B, N], it will be contiguous after transposed.